### ISSUE TITLE
Question about cross-domain generalization and classification head in TSGNN

### ISSUE THREAD

**Issue by YanJiangJerry** [benfinkelshtein/EquivarianceEverywhere#1]
Hi, thanks for the great work!

I was wondering whether TSGNN is designed to support cross-domain generalization — that is, pretraining on one dataset and testing on another.

If so, how does TSGNN handle cases where the number of classes differs between the pretraining and testing datasets? Specifically, how can the classification head be trained during pretraining if the test-time classes are unseen or have a different dimension?

I have run the script you provided, and it seems that the training, validation, and test graphs are all from the same domain -- cora:
`[Train]      x: torch.Size([2708, 1433]), y_mat: torch.Size([2708, 7]), y: torch.Size([2708]), train_mask: 140  
[Test-val]   x: torch.Size([2708, 1433]), y_mat: torch.Size([2708, 7]), y: torch.Size([2708]), val_mask: 500  
[Test-test]  x: torch.Size([2708, 1433]), y_mat: torch.Size([2708, 7]), y: torch.Size([2708]), test_mask: 1000
`

I am also curious about whether the TSGNN need all the labels from the target graph so that it can be adapted. **If I train a TSGNN on cora and test on texas, the only information I need is the full labelled set from cora and I don't need to know any information (such as class number and labelled samples) from texas, it that correct?**

Looking forward to your clarification!
